tests/e2e/framework/kubernetes.go

```diff
@@ -138,11 +139,29 @@ func (k *KubeInfo) Teardown() error {
 	var err error
 	if k.namespaceCreated {
 		if err = util.DeleteNamespace(k.Namespace); err != nil {
-			glog.Error("Failed to delete namespace")
+			glog.Errorf("Failed to delete namespace %s", k.Namespace)
+			return err
+		}
+
+		// confirm the namespace is deleted as it will cause future creation to fail
+		maxAttempts := 15
```

Comment:
cool, thanks for pointing that out, that is good to know!  The only thing i didn't like about the retrier is that it is random vs keep increasing the delay time which i think makes more sense.